ledger/participant-integration-api/src/main/scala/platform/index/InMemoryStateUpdater.scala

```diff
@@ -100,14 +93,42 @@ private[platform] object InMemoryStateUpdater {
     updateCachesExecutionContext = ExecutionContext.fromExecutorService(updateCachesExecutor),
     metrics = metrics,
   )(
-    updateCaches = updateCaches(inMemoryState),
-    convertTransactionAccepted = convertTransactionAccepted,
-    convertTransactionRejected = convertTransactionRejected,
-    updateLedgerEnd = updateLedgerEnd(inMemoryState),
+    prepare = prepare,
+    update = update(inMemoryState, loggingContext),
   )
 
-  private def updateCaches(inMemoryState: InMemoryState)(
-      updates: Vector[TransactionLogUpdate]
+  def combinePackageMetadata(batch: Vector[(Offset, Update)]): PackageMetadata =
```

Comment:
naming nit: naming suggest in this form this is about some aggregation, but in fact we decode and do the heavy lifting in the middle, can we reflect this in naming somehow?